start.py

```diff
@@ -1061,3 +1067,26 @@ def failure(timeout):
     wait_for_condition(condition=condition, condition_args=[deployment, cluster_name],
                        time_between_checks=3, timeout=600, time_to_success=30,
                        success=success, failure=failure)
+
+
+def _execute_commands_against_kerberized_hdfs(cluster, commands, quiet):
+    hdfs_principal = 'hdfs/{hdfs_node_name}@{realm}'.format(hdfs_node_name=cluster.primary_node.fqdn,
+                                                            realm=cluster.network.upper())
+    kinit_command = [('kinit -k -t "$(find /var/run/cloudera-scm-agent/process/*hdfs-NAMENODE '
+                      '-maxdepth 0 -mindepth 0 | sort -rs | head -1)/hdfs.keytab" {}').format(hdfs_principal)]
+    kdestroy_command = ['kdestroy']
+    cluster.primary_node.execute(' && '.join(kinit_command + commands + kdestroy_command), quiet=quiet)
+
+
+def _configure_after_start(deployment, cluster_name, cluster, quiet, args):
```

Comment:
Instead of passing `args`, I'd just add an arg for `kerberos_principals`. When I create functions, I think it helps to have the argument list be somewhat clear about how the arguments might be used in the body (just passing `args`, I wouldn't know why you might want that whole collection until I read into it).